src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -34,17 +34,21 @@
 public final class HybridQuery extends Query implements Iterable<Query> {
 
     private final List<Query> subQueries;
+    private final HybridQueryContext queryContext;
 
     /**
      * Create new instance of hybrid query object based on collection of sub queries and filter query
      * @param subQueries collection of queries that are executed individually and contribute to a final list of combined scores
      * @param filterQueries list of filters that will be applied to each sub query. Each filter from the list is added as bool "filter" clause. If this is null sub queries will be executed as is
      */
-    public HybridQuery(final Collection<Query> subQueries, final List<Query> filterQueries) {
+    public HybridQuery(final Collection<Query> subQueries, final List<Query> filterQueries, final HybridQueryContext hybridQueryContext) {
         Objects.requireNonNull(subQueries, "collection of queries must not be null");
         if (subQueries.isEmpty()) {
             throw new IllegalArgumentException("collection of queries must not be empty");
         }
+        if (hybridQueryContext.getPaginationDepth() == 0) {
```

Comment:
Should we call the `validatePaginationDepth` method in `HybridQueryBuilder` here since there are more restrictions than not being 0?

Reply:
This is the validation to set the default value in case where user didn't send pagination depth in query and we add it a default value from the system.

Reply:
We already have a validate pagination depth method in doToQuery method where on the basis of paginationdepth we evaluate whether to throw exception or move forward.

Reply:
Makes sense, approving